dynawo/sources/Solvers/AlgebraicSolvers/DYNSolverKINAlgRestoration.cpp

```diff
@@ -124,12 +118,12 @@ SolverKINAlgRestoration::initVarAndEqTypes() {
   switch (mode_) {
     case KIN_NORMAL: {
       for (int i = 0; i < model_->sizeF(); ++i) {
-        if (fType_[i] > 0)
+        if (modelFType[i] > 0)
           ignoreF_.insert(i);
         else
           indexF_.push_back(i);
 
-        if (vId_[i] > 0)
+        if (modelYType[i] > 0)
           ignoreY_.insert(i);
         else
           indexY_.push_back(i);
```

Comment:
I get the impression that we can merge a bit more code to only have one for loop for calculating numF and filling ignoreF/ignoreY, which means that instead of using a count above to have numF. We do it manually in this for loop, the changes that numF being equal to zero being quite zero ;)
It will also make the assert on numF and indexF useless as this will be guaranteed by construction.

Reply:
I changed like you said. But assert is on indexY and numF so it should be kept, assert are only in debug so the most we use them the better